### DISCUSSION TITLE
Serverless node.js helpers not working

### DISCUSSION THREAD

**Discussion by jlrodvPF** [vercel/community#6897] [Help] [answered]
### Summary

Im creating a serverless function and Im trying to use the query helper function but if I try to print request.query it prints undefined and I can see "Property 'query' does not exist on type 'Request'."

My function looks like this: 

`export async function GET(request: Request) {
  console.log(request. query)
}`

and  already ran npm install @vercel/node --save-dev 

am I missing an import or something like that?

### Steps to Reproduce

1.- Create a serverless function with node. 
2.- Try to print request.query
3.- It prints undefined

**Comment by QuiiBz** [comment, marked as answer]
The `query` property indeed doesn't exist on the `Request` object: https://developer.mozilla.org/en-US/docs/Web/API/Request

I'm assuming you want to access the query parameters of the request. You can do so by accessing the `url` property, constructing a new [`URL` object](https://developer.mozilla.org/en-US/docs/Web/API/URL) and accessing the `searchParams` property:

```ts
export async function GET(request: Request) {
  const url = new URL(request.url)
  const query = url.searchParams
  // query is a URLSearchParams object: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
}
```

**Comment by jlrodvPF** [comment]
Understand that but I expected to find the .query from Vercel helpers https://vercel.com/blog/vercel-node-helpers

**Reply by QuiiBz** [reply]
Helpers are made to work on the syntax for Serverless Functions, which is:

```ts
module.exports = (request, response) => {
  ...
}
```

The new syntax is:

```ts
export async function GET(request) {
   ...
}
```

You can still use both but we recommend the latter.

**Reply by jlrodvPF** [reply]
Oh ok understood, maybe docs need to be updated because I see the new syntax in the quick start https://vercel.com/docs/functions/quickstart and when moving to node.js section https://vercel.com/docs/functions/runtimes/node-js there's no mention about the new and old syntax and doesn't mention that the helpers only work in the old syntax . Anyways thanks for your help!
